### ISSUE TITLE
Move all internal auth management and usage to an external shared package available on CRAN

### ISSUE THREAD

**Issue by jdeboer** [jdeboer/ganalytics#28]
Utilise the existing [googleAuthR](https://github.com/MarkEdmondson1234/googleAuthR) package by @MarkEdmondson1234 where possible. This way, the ganalytics package can be modularised into smaller packages, some of which could be reused by other Google Analytics related R packages.

**Comment by MarkEdmondson1234** [comment]
Let me know if I can help! And not just GA related packages, any Google API that is made with googleAuthR is the ambition, so things like BigQuery, PredictionAPI, GoogleDrive etc. etc.

**Comment by jdeboer** [comment]
Yes, that would be great. I think there will be some functionality which there may not directly match, but we can work on solving those together. How would you like to go about this?

**Comment by MarkEdmondson1234** [comment]
I didn't write googleAuthR with integration into existing packages in mind, so I may need to expose some inner functions, but otherwise you just need to replace where you call the Google OAuth2 token (the one you use with the request) with the googleAuthR `Authentication$public_fields$token`, then users authenticate via `googleAuthR::gar_auth()`

For Shiny compatibility, it may be a bit more involved, but I suggest the above working first.

**Comment by jdeboer** [comment]
Hi @MarkEdmondson1234 , the approach I'm thinking to take with this is to firstly allow the use of the `googleAuthR` as an alternative means for users to authenticate while still providing the existing authentication methods available in `ganalytics`. Once that's done, we can then look for ways to begin replacing those existing authentication methods in a backwards compatible way such as by wrapping `googleAuthR` functions inside the `ganalytics` counterparts.
